timemachine/potentials/nonbonded.py

```diff
@@ -740,13 +740,13 @@ def lj_interaction_group_energy(sig_ligand, eps_ligand, lj_prefactors):
 
     Parameters
     ----------
-    sig_ligand, eps_ligand: [N_lig] arrays
-    lj_prefactors: [N_lig, 20] array
+    sig_ligand, eps_ligand: [N_frames, N_lig] arrays
+    lj_prefactors: [N_frames, N_lig, 20] array
 
     Returns
     -------
-    energy: float
+    energy: [N_frames] array
     """
 
     projection = vmap(basis_expand_lj_atom)(sig_ligand, eps_ligand)
-    return jnp.sum(projection * lj_prefactors)
+    return jnp.sum(jnp.sum(projection * lj_prefactors, axis=2), axis=1)
```

Comment:
Preference: leave this function as defined for a single snapshot, and add a function to apply it to a sequence of snapshots.

(Or update "Assuming lj_prefactors = lj_prefactors_on_snapshot" from docstring to match revised interface.)

Reply:
reverted

I see. I was relying on a feature of jnp.dot which made coulomb_interaction_group_energy happen to work on a trajectory (If a is an N-D array and b is a 1-D array, it is a sum product over the last axis of a and b). I'll update the calling code and keep this the same.